functests/1_performance/cpu_management.go

```diff
@@ -349,6 +311,76 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			}
 		})
 	})
+
+	Describe("Verification that IRQ load balance can be disabled per POD", func() {
+		var testpod *corev1.Pod
+
+		BeforeEach(func() {
+			if profile.Spec.GloballyDisableIrqLoadBalancing != nil && *profile.Spec.GloballyDisableIrqLoadBalancing {
+				Skip("IRQ load balance should be enabled (GloballyDisableIrqLoadBalancing=false), skipping test")
+			}
+		})
+
+		AfterEach(func() {
+			deleteTestPod(testpod)
+		})
+
+		It("[test_id:36364] should disable IRQ balance for CPU where POD is running", func() {
+			By("checking default smp affinity is equal to all active CPUs")
+			defaultSmpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(defaultSmpAffinitySet).To(Equal(onlineCPUsSet), fmt.Sprintf("Default SMP Affinity %s should be equal to all active CPUs %s", defaultSmpAffinitySet, onlineCPUsSet))
+
+			By("Running pod with annotations that disable specific CPU from IRQ balancer")
+			annotations := map[string]string{
+				"irq-load-balancing.crio.io": "true",
+				"cpu-quota.crio.io":          "true",
+			}
+			testpod = getTestPodWithAnnotations(annotations)
+
+			err = testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+			runTop := []string{"/bin/bash", "-c", "tmux new -d 'top -b > /dev/null'"}
+			_, err = pods.ExecCommandOnPod(testpod, runTop)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking that the default smp affinity mask was updated and CPU (where 'top' is running) isolated")
+			defaultSmpAffinitySet, err = nodes.GetDefaultSmpAffinitySet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			getPsr := []string{"/bin/bash", "-c", "ps -o psr $(pgrep top) | tail -1"}
+			psr, err := nodes.ExecCommandOnNode(getPsr, workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+			psrSet, err := cpuset.Parse(strings.Trim(psr, " "))
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(defaultSmpAffinitySet).To(Equal(onlineCPUsSet.Difference(psrSet)), fmt.Sprintf("Default SMP affinity should not contain isolated CPU %s", psr))
+
+			By("Checking that there are no any active IRQ on isolated CPU")
+			// It may takes some time for the system to reschedule active IRQs
+			Eventually(func() string {
+				getActiveIrq := []string{"/bin/bash", "-c", "for n in $(find /proc/irq/ -name smp_affinity_list); do echo $(cat $n); done"}
+				activeIrqList, err := nodes.ExecCommandOnNode(getActiveIrq, workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+				re := regexp.MustCompile(fmt.Sprintf(`(?m)^%s$`, psrSet))
+				return re.FindString(activeIrqList)
```

Comment:
Nice! have you seen this fail for a pod without the annotations?

Reply:
yes, here the output for the pod without annitation:


```
    Timed out after 30.000s.
    IRQ still active on CPU  1
    Expected
        <string>: 1
    to be empty

```

Reply:
Nice!